File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    import math
    prime_arr = [True] * (n+1)
    prime_arr[0] = [False]
    prime_arr[1] = [False]
    for i in range(2, int(math.sqrt(n)+1)):
        if prime_arr[i] == True:
            j = 2
            while ( i * j ) <= n:
                prime_arr[i*j] = False
                j += 1
    return prime_arr.count(True)
            
                    
                

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int count = scanner.nextInt();
        scanner.nextLine();

        String[] arr = scanner.nextLine().split(" ");
        int[] numbers = new int[arr.length];
        for(int i = 0; i < numbers.length; i++) {
            numbers[i] = Integer.parseInt(arr[i]);
        }

        int result = calculate(numbers);

        System.out.println(result);

    }

    public static int calculate(int[] numbers) {
        int count = 0;

        for(int i = 0; i < numbers.length; i++) {
            if(isPrime(numbers[i])) {
                count += 1;
            }
        }
        return count;
    }

    public static boolean isPrime(int num) {
        if (num < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}